src/client/java/teammates/client/scripts/sql/VerifyCourseEntityAttributes.java

```diff
@@ -0,0 +1,150 @@
+
+    public VerifyCourseEntityAttributes() {
+        super(Course.class,
+                teammates.storage.sqlentity.Course.class);
+    }
+
+    @Override
+    protected String generateID(teammates.storage.sqlentity.Course sqlEntity) {
+        return sqlEntity.getId();
+    }
+
+    public static void main(String[] args) {
+        VerifyCourseEntityAttributes script = new VerifyCourseEntityAttributes();
+        script.doOperationRemotely();
+    }
+
+    // Used for sql data migration
+    @Override
+    public boolean equals(teammates.storage.sqlentity.Course sqlEntity, Course datastoreEntity) {
+        try {
+
+            return verifyCourse(sqlEntity, datastoreEntity) && verifySectionChain(sqlEntity);
+        } catch (IllegalArgumentException iae) {
+            return false;
+        }
+    }
+
+    private boolean verifyCourse(teammates.storage.sqlentity.Course sqlEntity, Course datastoreEntity) {
+        return sqlEntity.getId().equals(datastoreEntity.getUniqueId())
+                && sqlEntity.getName().equals(datastoreEntity.getName())
+                && sqlEntity.getTimeZone().equals(datastoreEntity.getTimeZone())
+                && sqlEntity.getInstitute().equals(datastoreEntity.getInstitute())
+                // && sqlEntity.getCreatedAt().equals(datastoreEntity.getCreatedAt())
+                && datastoreEntity.getDeletedAt() == null ? sqlEntity.getDeletedAt() == null
+                        : sqlEntity.getDeletedAt().equals(datastoreEntity.getDeletedAt());
+    }
+
+    private boolean verifySectionChain(teammates.storage.sqlentity.Course sqlEntity) {
+        List<CourseStudent> oldStudents = ofy().load().type(CourseStudent.class).filter("courseId", sqlEntity.getId())
+                .list();
+
+        HibernateUtil.beginTransaction();
+        CriteriaBuilder cb = HibernateUtil.getCriteriaBuilder();
+        CriteriaQuery<teammates.storage.sqlentity.Student> cr = cb
+                .createQuery(teammates.storage.sqlentity.Student.class);
+        Root<teammates.storage.sqlentity.Student> courseRoot = cr.from(teammates.storage.sqlentity.Student.class);
+        cr.select(courseRoot).where(cb.equal(courseRoot.get("courseId"), sqlEntity.getId()));
+        List<Student> newStudents = HibernateUtil.createQuery(cr).getResultList();
+        HibernateUtil.commitTransaction();
+
+        Map<String, List<CourseStudent>> sectionToOldStuMap = oldStudents.stream()
+                .collect(Collectors.groupingBy(CourseStudent::getSectionName));
+        Map<String, List<Student>> sectionToNewStuMap = newStudents.stream()
+                .collect(Collectors.groupingBy(Student::getSectionName));
+
+        List<Section> newSection = sqlEntity.getSections();
+
+        // Enforce that they are equal in size so the next check can check if all
+        // sections are present
+        if (newSection.size() != sectionToOldStuMap.size() || newSection.size() != sectionToNewStuMap.size()) {
```

Comment:
Can do something similar to this, to make use of code as documentation, and do away with the comments. This way, we can see more clearly the parts where the count check was made too.

```suggestion
        boolean isSectionsCountEqual = (newSection.size() != sectionToOldStuMap.size() || newSection.size() != sectionToNewStuMap.size());
        if (!isSectionsCountEqual) {
            return false;
        }
```

Reply:
Sure good catch!